src/versions/validation/event.js

```diff
@@ -10,10 +10,56 @@ const orderByFields = [
 
 const schemas = {}
 
+const getObjectSchema = (name) => Joi.string().regex(new RegExp(`^${name}\\.\\w+$`))
+
+const groupBySchema = Joi.alternatives().try(
+  Joi.string().valid(
+    'type',
+    'objectType',
+    'objectId',
+    'parentId',
+    'emitter',
+    'emitterId'
+  ),
+  getObjectSchema('object'),
+  getObjectSchema('metadata')
+)
+
+const fieldSchema = Joi.alternatives().try(
+  getObjectSchema('object'),
+  getObjectSchema('metadata')
+)
+
 // ////////// //
 // 2019-05-20 //
 // ////////// //
 schemas['2019-05-20'] = {}
+schemas['2019-05-20'].getStats = {
+  query: Joi.object().keys({
+    // order
+    orderBy: Joi.string().valid('avg', 'count', 'sum', 'min', 'max').default('count'),
+    order: Joi.string().valid('asc', 'desc').default('desc'),
+
+    // pagination
+    page: Joi.number().integer().min(1).default(1),
+    nbResultsPerPage: Joi.number().integer().min(1).max(100).default(DEFAULT_NB_RESULTS_PER_PAGE),
+
+    // aggregation
+    groupBy: groupBySchema.required(),
+    field: fieldSchema,
+    avgPrecision: Joi.number().integer().min(0).default(2),
+
+    // filters
+    id: [Joi.string(), Joi.array().unique().items(Joi.string())],
+    createdDate: getRangeFilter(Joi.string().isoDate()),
+    type: [Joi.string(), Joi.array().unique().items(Joi.string())],
+    objectType: [Joi.string(), Joi.array().unique().items(Joi.string())],
+    objectId: [Joi.string(), Joi.array().unique().items(Joi.string())],
```

Comment:
Why not simply coerce using `Joi.array().unique()`?

Reply:
Right, thanks!